include/Standalone/Transforms.h

```diff
@@ -29,10 +29,23 @@ namespace linalgx {
 FailureOr<SmallVector<Value>> mapToBRGEMMOp(RewriterBase &rewriter,
                                             linalg::LinalgOp linalgOp);
 
-//
-FailureOr<linalg::MatmulOp> mapConvToGemm(RewriterBase &rewriter,
-                                          linalg::LinalgOp linalgOp,
-                                          int64_t rPos, int64_t sPos);
+// Map a convolution to a matmul operation.
+// [N][P][Q][K] += [N][H][W][C] * [R][S][C][K]
+// [N][K’][P][Q][k] += [N][C’][H][W][c] * [K’][C’][R][S][c][k] (blocked)
+class MapConvToMatmul {
+public:
+  MapConvToMatmul() = delete;
+  MapConvToMatmul(int64_t rPos, int64_t sPos) : rPos(rPos), sPos(sPos){};
+  FailureOr<linalg::MatmulOp> doIt(RewriterBase &rewriter,
+                                   linalg::LinalgOp linalgOp);
+
+  int64_t getRPos() { return rPos; };
+  int64_t getSPos() { return sPos; };
+
+private:
+  int64_t rPos;
```

Comment:
isn't R and S always contiguous? Isn't the C position what really matters here? Perhaps this should have a single argument, and derive the others from it.

Reply:
Why the channel? I am interested in the filter width and height as I need them to understand the matmul sizes. They are contiguous so we could specify only `Rpos` or `sPos`.

Reply:
Perhaps pass both and just have an assert that `s = r+1`?

Reply:
Will do in a follow-up